pkg/app/daemon/daemon.go

```diff
@@ -94,19 +106,52 @@ func (self *rebaseDaemon) Run() error {
 }
 
 func (self *rebaseDaemon) writeTodoFile(path string) error {
-	todoContent := []byte(os.Getenv(RebaseTODOEnvKey))
+	if changeTodoActionEnvValue := os.Getenv(ChangeTodoActionEnvKey); changeTodoActionEnvValue != "" {
+		return self.changeTodoAction(path, changeTodoActionEnvValue)
+	} else {
+		todoContent := []byte(os.Getenv(RebaseTODOEnvKey))
+
+		prependLines := os.Getenv(PrependLinesEnvKey) != ""
+		if prependLines {
+			existingContent, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			todoContent = append(todoContent, existingContent...)
+		}
+
+		return os.WriteFile(path, todoContent, 0o644)
+	}
+}
 
-	prependLines := os.Getenv(PrependLinesEnvKey) != ""
-	if prependLines {
-		existingContent, err := os.ReadFile(path)
-		if err != nil {
+func (self *rebaseDaemon) changeTodoAction(path string, changeTodoActionEnvValue string) error {
+	lines := strings.Split(changeTodoActionEnvValue, "\n")
+	for _, line := range lines {
+		fields := strings.Split(line, ":")
+		if len(fields) != 2 {
+			return fmt.Errorf("Unexpected value for %s: %s", ChangeTodoActionEnvKey, changeTodoActionEnvValue)
+		}
+		sha, newAction := fields[0], self.actionFromString(fields[1])
+		if int(newAction) == 0 {
+			return fmt.Errorf("Unknown action in %s", changeTodoActionEnvValue)
+		}
+		if err := utils.EditRebaseTodo(path, sha, todo.Pick, newAction); err != nil {
 			return err
 		}
+	}
 
-		todoContent = append(todoContent, existingContent...)
+	return nil
+}
+
+func (self *rebaseDaemon) actionFromString(actionString string) todo.TodoCommand {
+	for t := todo.Pick; t < todo.Comment; t++ {
+		if t.String() == actionString {
+			return t
+		}
 	}
 
-	return os.WriteFile(path, todoContent, 0o644)
+	return 0
```

Comment:
Rather than use 0 as a sentinel value, I'd have the function return `(todo.TodoCommand, bool)` where a false value means we couldn't find a match. That defends us against the situation where 0 becomes a valid value in that enum. In general I don't like depending on the underlying values of a 3rd party enum cos it's subject to change and it's hard to reason about without going and looking at the enum definition.

While I'm on the topic, I wonder if we should be mapping from the todo package's todo type to our own in a central location to reduce our exposure to this stuff. I don't know how much effort that would be, I'm happy to be told it's overkill.

Reply:
Also obsolete now.

As for the question of creating our own enum and mapping it to the todo package's todo type, I did consider this when I turned `Commit.Action` into an enum. We would have been forced to do that if `todo.TodoCommand` wasn't one-based. I decided that it's not necessary because it's so unlikely that the todo package will suddenly change its enum to start at 0. And it can be even more error-prone, because the mapping would have to be maintained manually, so if the todo package's enum changes (as it recently did in https://github.com/fsmiamoto/git-todo-parser/pull/2), we might not notice and end up with bogus mappings.